src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidContentProviderBroker.cs

```diff
@@ -0,0 +1,349 @@
+            var OperationBundleJSON = _brokerHelper.SearializeBundleToJSON(silentOperationBundle);
+            var SilentOperationBundle = await PerformContentResolverOperationAsync(ContentResolverOperation.acquireTokenSilent, OperationBundleJSON).ConfigureAwait(false);
+            return _brokerHelper.GetSilentResultFromBundle(SilentOperationBundle);
+        }
+
+        public async Task<IEnumerable<IAccount>> GetAccountsAsync(string clientID, string redirectUri)
+        {
+            using (_logger.LogMethodDuration())
+            {
+                if (!IsBrokerInstalledAndInvokable())
+                {
+                    _logger.Warning("Android broker is either not installed or is not reachable so no accounts will be returned. ");
+                    return new List<IAccount>();
+                }
+
+                return await PerformGetAccountsAsync(clientID, redirectUri).ConfigureAwait(false);
+            }
+        }
+
+        private async Task<IEnumerable<IAccount>> PerformGetAccountsAsync(string clientID, string redirectUri)
+        {
+            BrokerRequest brokerRequest = new BrokerRequest() { ClientId = clientID, RedirectUri = new Uri(redirectUri) };
+
+            try
+            {
+                InitiateBrokerHandshakeAsync();
+                var accountData = await GetBrokerAccountDataAsync(brokerRequest).ConfigureAwait(false);
+                return _brokerHelper.ExtractBrokerAccountsFromAccountData(accountData);
+            }
+            catch (Exception ex)
+            {
+                _logger.Error("Failed to get Android broker accounts from the broker. ");
+                _brokerHelper.HandleBrokerOperationError(ex);
+                throw;
+            }
+        }
+
+        private async Task<string> GetBrokerAccountDataAsync(BrokerRequest brokerRequest)
+        {
+            var getAccountsBundle = _brokerHelper.CreateBrokerAccountBundle(brokerRequest);
+            var OperationBundleJSON = _brokerHelper.SearializeBundleToJSON(getAccountsBundle);
+            var bundleResult = await PerformContentResolverOperationAsync(ContentResolverOperation.getAccounts, OperationBundleJSON).ConfigureAwait(false);
+
+            return bundleResult?.GetString(BrokerConstants.BrokerAccounts);
+        }
+
+        public async Task RemoveAccountAsync(IApplicationConfiguration applicationConfiguration, IAccount account)
+        {
+            using (_logger.LogMethodDuration())
+            {
+                if (!IsBrokerInstalledAndInvokable())
+                {
+                    _logger.Warning("Android broker is either not installed or not reachable so no accounts will be removed. ");
+                    return;
+                }
+
+                await PerformRemoveAccountsAsync(applicationConfiguration, account).ConfigureAwait(false);
+            }
+        }
+
+        private async Task PerformRemoveAccountsAsync(IApplicationConfiguration applicationConfiguration, IAccount account)
```

Comment:
nit: generally I prefer the naming `RemoveAccountsInternalAsync()` or `RemoveAccountsImplAsync()` - the reasoning is that it keeps this drop down nicely grouped. I saw this naming patter in the C++ team code.

![image](https://user-images.githubusercontent.com/12273384/106802334-b84bd780-665a-11eb-8591-9c98bd25db31.png)
![image](https://user-images.githubusercontent.com/12273384/106802354-c1d53f80-665a-11eb-9733-64fb8a3be174.png)

Reply:
agree w/bogdan, also i would remove this method, it's not really doing anything and it's not being test separately, so just move the try/catch into RemoveAccountAsync

---
In reply to: [569710387](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2380#discussion_r569710387) [](ancestors = 569710387)